tests/test_seek_after_eos.c

```diff
@@ -71,6 +71,13 @@ int main(int ac, char **av)
                 nmd_frame_releasep(&frame);
             }
 
+            ret = nmd_get_next_frame(s, &frame);
+            if (ret != NMD_ERR_EOF) {
+                fprintf(stderr, "expected to be at EOF\n");
+                ret = -1;
+                goto done;
+            }
```

Comment:
That would'nt work. It would only makes sense for the case where `k == 2`. I don't think it defeats the purpose of the different frame pulling mechanism as we are either calling seek() or using the `get_frame()`. The chunk I added is only there to ensure we are actually at EOF (as the api mandates).